src/Xenial.Tasty.Cli/TastyServer.cs

```diff
@@ -0,0 +1,216 @@
+
+        public void SignalEndTestPipeline()
+            => EndTestPipelineSignaled?.Invoke();
+
+        public void SignalTestPipelineCompleted()
+            => TestPipelineCompletedSignaled?.Invoke();
+
+        public void ClearConsole()
+        {
+            try
+            {
+                Console.Clear();
+            }
+            catch (IOException) { /* Handle is invalid */}
+        }
+
+        public static class ConsoleReporter
+        {
+            public static ColorScheme Scheme = ColorScheme.Default;
+
+            static ConsoleReporter()
+                => Console.OutputEncoding = Encoding.UTF8;
+
+
+            static Lazy<int> SeparatorSize = new Lazy<int>(() =>
+            {
+                const int fallBackSize = 100;
+                try
+                {
+                    return Console.BufferWidth;
+                }
+                catch (IOException) { /* Handle is invalid */ }
+                return fallBackSize;
+            });
+
+            public static Task ReportSummary(IEnumerable<SerializableTestCase> tests)
+            {
+                var totalTests = tests.Count();
+                var failedTests = tests.Where(m => m.TestOutcome == TestOutcome.Failed).Count();
+                var ignoredTests = tests.Where(m => m.TestOutcome == TestOutcome.Ignored).Count();
+                var inconclusiveTests = tests.Where(m => m.TestOutcome == TestOutcome.NotRun).Count();
+                var successTests = tests.Where(m => m.TestOutcome == TestOutcome.Success).Count();
+                var outcome = tests.Where(t => t.TestOutcome > TestOutcome.Ignored).Min(t => t.TestOutcome);
+
+                var totalTime = tests.Sum(m => m.Duration);
+                var failedTime = tests.Where(m => m.TestOutcome == TestOutcome.Failed).Sum(m => m.Duration);
+                var ignoredTime = tests.Where(m => m.TestOutcome == TestOutcome.Ignored).Sum(m => m.Duration);
+                var inconclusiveTime = tests.Where(m => m.TestOutcome == TestOutcome.NotRun).Sum(m => m.Duration);
+                var successTime = tests.Where(m => m.TestOutcome == TestOutcome.Success).Sum(m => m.Duration);
+
+                var totalTimeString = totalTime.AsDuration();
+
+                Console.WriteLine();
+                Console.WriteLine(new string('=', SeparatorSize.Value));
+
+                Write(Scheme.DefaultColor, $"Summary: ");
+                Write(failedTests > 0 ? Scheme.ErrorColor : Scheme.DefaultColor, $"F{failedTests}".PadLeft(totalTimeString.Length));
+                Write(Scheme.DefaultColor, $" | ");
+                Write(ignoredTests > 0 ? Scheme.WarningColor : Scheme.DefaultColor, $"I{ignoredTests}".PadLeft(totalTimeString.Length));
+                Write(Scheme.DefaultColor, $" | ");
+                Write(inconclusiveTests > 0 ? Scheme.NotifyColor : Scheme.DefaultColor, $"NR{inconclusiveTests}".PadLeft(totalTimeString.Length));
+                Write(Scheme.DefaultColor, $" | ");
```

Comment:
NR => not run?
maybe rename variable then

Reply:
Good Catch! Yeah it's not run, i ditched inconclusive, cause there is ignored and would cause even more confusion.

Reply:
Done